docs: git providers section textual and visual improvements

stefanicjuraj opened the pull request:
- added textual additions and improvements per each sub-section
- added visual outputs when executing the cli commands with explanations
- organized the structure of content
- all changes have been tested and previewed locally

Tpuljak commented on src/content/docs/configuration/git-providers.mdx:
```diff
@@ -10,40 +10,82 @@ Daytona allows you to configure Git Providers and use repositories hosted on the
 ## Add a Git Provider
 
-You can add a Git Provider to Daytona.
+You can add a Git Provider to Daytona to integrate your version control system with your development workflow. This integration allows Daytona to interact with your repositories, enabling automated processes and continuous integration.
```
`, enabling automated processes and continuous integration.`

This is a bit ambitious at this stage 😄 

I suggest removing it. @ivan-burazin agree?

ivan-burazin replied:
I would be more to the point, something like

```
Daytona allows you to integrate your preferred Git provider, enhancing your development workflow. This integration enables you to access and manage your repositories directly within Daytona, simplifying workspace creation from existing projects.
Also by adding a Git provider, your credentials are securely embedded into your workspaces. This allows for seamless Git operations without repeated authentication.
```

Tpuljak reviewed (changes requested):
We should also add AWS CodeCommit and Azure DevOps to the docs. I'll add separate issues for that.

stefanicjuraj commented:
- added azure devops git provider (fixes https://github.com/daytonaio/docs/issues/108)
- added aws codecommit git provider (fixes https://github.com/daytonaio/docs/issues/109)
- added a git provider addition process for each provider (fixes https://github.com/daytonaio/docs/issues/110)
- modified descriptions

Tpuljak commented:
> Let me know if you would like to keep a general add git provider process (https://github.com/daytonaio/docs/commit/f3e69d5262adfbbbc4dbe17971257ad51f2746b0).

I would actually prefer to keep that

@ivan-burazin thoughts?

ivan-burazin commented:
> > Let me know if you would like to keep a general add git provider process ([f3e69d5](https://github.com/daytonaio/docs/commit/f3e69d5262adfbbbc4dbe17971257ad51f2746b0)).
> 
> I would actually prefer to keep that (in addition to the git-provider-specific steps)
> 
> @ivan-burazin thoughts?

Yeah I think we can keep a short and generic connection process, and them say for specific providers scroll below.

Also feel that: Connect, should be close to [List Connected Git Providers] &
[Delete a Git Provider]

Lastly i would remove this prerequisite from all pages, i feel like you should know this

```
Prerequisite

A working installation of Daytona

Verify your Daytona installation is up-to-date to avoid compatibility issues.
```

Tpuljak reviewed (approved):
Nice work!

Tpuljak commented on src/content/docs/configuration/git-providers.mdx:
```diff
@@ -4,148 +4,647 @@ description: Connect Daytona with GitHub, GitLab, Bitbucket, and other Git Provi
-        Select if you're hosting repositories on GitLab.com.
-    * __[Bitbucket](#bitbucket)__  
-        Select if you're hosting repositories on Atlassian's Bitbucket Cloud offering.
-    * __Other__
```
Disregard 😄
